templates/base/client/.hathora/client.ts.hbs

```diff
@@ -15,109 +17,61 @@ import {
   {{/each}}
   HathoraEvents,
 } from "../../api/types";
-import { UserData, Response, Method, COORDINATOR_HOST, MATCHMAKER_HOST } from "../../api/base";
+import { UserData, Response, Method } from "../../api/base";
 
 import { computePatch } from "./patch";
 import { ConnectionFailure, transformCoordinatorFailure } from "./failures";
 
-export type StateId = string;
-export type UpdateArgs = { stateId: StateId; state: UserState; updatedAt: number; events: HathoraEvents[] };
-export type UpdateCallback = (updateArgs: UpdateArgs) => void;
-export type ErrorCallback = (error: ConnectionFailure) => void;
-
-export class HathoraClient {
-  public appId = "{{appId}}";
-  private _client = new _HathoraClient(this.appId, COORDINATOR_HOST);
-
-  public static getUserFromToken(token: string): UserData {
-    return _HathoraClient.getUserFromToken(token) as UserData;
-  }
-
-{{#each auth}}
-{{#if (eq @key "anonymous")}}
-  public async loginAnonymous(): Promise<string> {
-    return this._client.loginAnonymous();
-  }
-
-{{else if (eq @key "nickname")}}
-  public async loginNickname(nickname: string): Promise<string> {
-    return this._client.loginNickname(nickname);
-  }
-
-{{else if (eq @key "google")}}
-  public async loginGoogle(idToken: string): Promise<string> {
-    return this._client.loginGoogle(idToken);
-  }
-
-{{/if}}
-{{/each}}
-  public async create(token: string, request: IInitializeRequest): Promise<string> {
-    return this._client.create(token, IInitializeRequest.encode(request).toBuffer());
+export class HathoraClient extends _HathoraClient {
+  public public_appId = "{{appId}}";
+  constructor() {
+    const connectionInfo = {
+      host: "localhost",
+      port: 4000,
+      transportType: "tcp" as const,
```

Comment:
in production this needs to be `connectionInfo = undefined` i believe

Reply:
because the server may not be running on localhost

Reply:
what's the criterion for "production" here? it may still be on localhost if `NODE_ENV === 'production'` right? is there an env var to detect hathora cloud or something?